confd/pkg/backends/calico/client.go

```diff
@@ -1187,6 +1203,36 @@ func (c *client) getLogSeverityKVPair(v3res *apiv3.BGPConfiguration, key interfa
 	}
 }
 
+func (c *client) getNodeMeshRestartTimeKVPair(v3res *apiv3.BGPConfiguration, key interface{}) {
+	meshRestartKey := getBGPConfigKey("node_mesh_restart_time", key)
+
+	if v3res != nil && v3res.Spec.NodeMeshMaxRestartTime != nil {
+		restartTime := *v3res.Spec.NodeMeshMaxRestartTime
+		c.updateCache(api.UpdateTypeKVUpdated, getKVPair(meshRestartKey, fmt.Sprintf("%v", int(math.Round(restartTime.Duration.Seconds())))))
+	} else {
+		c.updateCache(api.UpdateTypeKVDeleted, getKVPair(meshRestartKey))
+	}
+}
+
+func (c *client) getNodeMeshPasswordKVPair(v3res *apiv3.BGPConfiguration, key interface{}) {
+	meshPasswordKey := getBGPConfigKey("node_mesh_password", key)
+
+	if c.secretWatcher != nil && v3res.Spec.NodeMeshPassword != nil && v3res.Spec.NodeMeshPassword.SecretKeyRef != nil {
+		password, err := c.secretWatcher.GetSecret(
+			v3res.Spec.NodeMeshPassword.SecretKeyRef.Name,
+			v3res.Spec.NodeMeshPassword.SecretKeyRef.Key,
+		)
+		if err != nil {
+			log.WithError(err).Warningf("Can't read node mesh password in BGP Configuration %v", v3res.Name)
```

Comment:
A bit confusing - maybe "can't read password secret referenced by BGP configuration"? And include the name of the secret?